src/Planners/src/SwingFootPlanner.cpp

```diff
@@ -0,0 +1,270 @@
+        return false;
+    }
+
+    if (!handler->getParameter("foot_apex_time", m_footApexTime))
+    {
+        std::cerr << "[SwingFootPlanner::initialize] Unable to initialize the foot apex time for "
+                     "the SwingFootPlanner.";
+        return false;
+    }
+
+    double footLandingVelocity;
+    if (!handler->getParameter("foot_landing_velocity", footLandingVelocity))
+    {
+        std::cerr << "[SwingFootPlanner::initialize] Unable to initialize the foot landing "
+                     "velocity for the SwingFootPlanner.";
+        return false;
+    }
+
+    double footLandingAcceleration;
+    if (!handler->getParameter("foot_landing_acceleration", footLandingAcceleration))
+    {
+        std::cerr << "[SwingFootPlanner::initialize] Unable to initialize the foot landing "
+                     "acceleration for the SwingFootPlanner.";
+        return false;
+    }
+
+    // check the parameters passed to the planner
+    bool ok = (m_dT > 0) && ((m_footApexTime > 0) && (m_footApexTime < 1));
+    if (!ok)
+    {
+        std::cerr << "[SwingFootPlanner::initialize] There is a problem in the parameters used to "
+                     "configure the planner. Please remember to set a strictly positive "
+                     "sampling_time, and a foot_apex_time strictly between 0 and 1."
+                  << std::endl;
+        return false;
+    }
+
+    m_planarPlanner.setInitialConditions(Eigen::Vector2d::Zero(), Eigen::Vector2d::Zero());
+    m_planarPlanner.setFinalConditions(Eigen::Vector2d::Zero(), Eigen::Vector2d::Zero());
+
+    m_heightPlanner.setInitialConditions(Vector1d::Zero(), Vector1d::Zero());
+    m_heightPlanner.setFinalConditions(Vector1d::Constant(footLandingVelocity),
+                                       Vector1d::Constant(footLandingAcceleration));
+
+    return true;
+}
+
+void SwingFootPlanner::setContactList(const ContactList& contactList)
+{
+    // reset the time
+    m_currentTrajectoryTime = 0;
+
+    m_contactList = contactList;
+
+    // set the first contact
+    m_currentContactPtr = m_contactList.firstContact();
+    m_state.transform = m_currentContactPtr->pose;
+    m_state.mixedVelocity.setZero();
+    m_state.mixedAcceleration.setZero();
+    m_state.isInContact = true;
+
+}
+
+bool SwingFootPlanner::isValid() const
+{
+    return m_contactList.size() != 0;
+}
+
+const SwingFootPlannerState& SwingFootPlanner::get() const
+{
+    return m_state;
+}
+
+bool SwingFootPlanner::updateSE3Traj()
+{
+    // compute the trajectory at the current time
+    const double shiftedTime = m_currentTrajectoryTime - m_currentContactPtr->deactivationTime;
+
+    manif::SO3d rotation;
+
+    // note here we assume that the velocity is expressed using the mixed representation.
+    // This is the only reason why we consider an element of the  tangent space of SE3 as a 6d
+    // vector containing linear and angular velocity. The same assumption holds also for the acceleration.
```

Comment:
by "an element" you mean to say the angular part?

Reply:
I rewrote the comment in: https://github.com/dic-iit/bipedal-locomotion-framework/pull/90/commits/1560108a3ec368b5aa9ebc3ef1cdea936456d6c3

Reply:
Thank you!